src/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.WriteProperties.DateTime.cs

```diff
@@ -0,0 +1,278 @@
+                    ValidateWritingProperty();
+                }
+                WriteStringIndented(ref propertyName, value);
+            }
+            else
+            {
+                if (!_writerOptions.SkipValidation)
+                {
+                    ValidateWritingProperty();
+                }
+                WriteStringMinimized(ref propertyName, value);
+            }
+        }
+
+        private void WriteStringByOptions(ref ReadOnlySpan<byte> propertyName, DateTime value)
+        {
+            if (_writerOptions.Indented)
+            {
+                if (!_writerOptions.SkipValidation)
+                {
+                    ValidateWritingProperty();
+                }
+                WriteStringIndented(ref propertyName, value);
+            }
+            else
+            {
+                if (!_writerOptions.SkipValidation)
+                {
+                    ValidateWritingProperty();
+                }
+                WriteStringMinimized(ref propertyName, value);
+            }
+        }
+
+        private void WriteStringMinimized(ref ReadOnlySpan<char> escapedPropertyName, DateTime value)
+        {
+            int idx = WritePropertyNameMinimized(ref escapedPropertyName);
+
+            WriteStringValue(value, ref idx);
+
+            Advance(idx);
+        }
+
+        private void WriteStringMinimized(ref ReadOnlySpan<byte> escapedPropertyName, DateTime value)
+        {
+            int idx = WritePropertyNameMinimized(ref escapedPropertyName);
+
+            WriteStringValue(value, ref idx);
+
+            Advance(idx);
+        }
+
+        private void WriteStringIndented(ref ReadOnlySpan<char> escapedPropertyName, DateTime value)
+        {
+            int idx = WritePropertyNameIndented(ref escapedPropertyName);
+
+            WriteStringValue(value, ref idx);
+
+            Advance(idx);
+        }
+
+        private void WriteStringIndented(ref ReadOnlySpan<byte> escapedPropertyName, DateTime value)
+        {
+            int idx = WritePropertyNameIndented(ref escapedPropertyName);
+
+            WriteStringValue(value, ref idx);
+
+            Advance(idx);
+        }
+
+        private void WriteStringValue(DateTime value, ref int idx)
+        {
+            while (_buffer.Length <= idx)
+            {
+                AdvanceAndGrow(idx);
+                idx = 0;
+            }
+            _buffer[idx++] = JsonConstants.Quote;
+
+            FormatLoop(value, ref idx);
+
+            while (_buffer.Length <= idx)
+            {
+                AdvanceAndGrow(idx);
+                idx = 0;
+            }
+            _buffer[idx++] = JsonConstants.Quote;
+        }
+
+        private void FormatLoop(DateTime value, ref int idx)
+        {
+            int bytesWritten;
+            while (!Utf8Formatter.TryFormat(value, _buffer.Slice(idx), out bytesWritten))
+            {
+                AdvanceAndGrow(idx, JsonConstants.MaximumDateTimeLength);
+                idx = 0;
```

Comment:
Is this correct?  It assumes that idx was passed in as 0, no?

Reply:
(Same for the other cases above where idx is set to 0)

Reply:
Ah, does AdvanceAndGrow flush, and idx is pointing into the buffer that was flushed?

Reply:
> does AdvanceAndGrow flush

Yes.

> idx is pointing into the buffer that was flushed?

I am not sure I understand the question. After buffer is flushed, idx is reset back to 0 to point to the start of the newly acquired buffer.